pymeasure/instruments/temptronic/temptronic_base.py

```diff
@@ -0,0 +1,820 @@
+    FLOW_SENSOR_HARDWARE_ERROR = 512    # bit 9  – flow sensor hardware error
+    # bit 8  – reserved
+    INTERNAL_ERROR = 128    # bit 7  – internal error
+    # bit 6  – reserved
+    AIR_SENSOR_OPEN = 32     # bit 5  – air sensor open
+    LOW_INPUT_AIR_PRESSURE = 16     # bit 4  – low input air pressure
+    LOW_FLOW = 8      # bit 3  – low flow
+    # bit 2  – reserved
+    AIR_OPEN_LOOP = 2      # bit 1  – air open loop
+    OVERHEAT = 1      # bit 0  – overheat
+    OK = 0  # ok state
+
+
+class ATSBase(Instrument):
+    """The base class for Temptronic ATSXXX instruments.
+    """
+
+    remote_mode = Instrument.setting(
+        "%s",
+        """``True`` disables TS GUI but displays a “Return to local" switch.""",
+        validator=strict_discrete_set,
+        values={True: "%RM", False: r"%GL"},
+        map_values=True
+    )
+
+    maximum_test_time = Instrument.control(
+        "TTIM?", "TTIM %g",
+        """Control maximum allowed test time (s).
+
+        :type: float
+
+        This prevents TS from staying at a single temperature forever.
+        Valid range: 0 to 9999
+        """,
+        validator=truncated_range,
+        values=[0, 9999]
+    )
+
+    dut_mode = Instrument.control(
+        "DUTM?", "DUTM %g",
+        """ ``On`` enables DUT mode, ``OFF`` enables air mode
+
+        :type: string
+
+        """,
+        validator=strict_discrete_set,
+        values={'ON': 1, 'OFF': 0},
+        map_values=True
+    )
+
+    dut_type = Instrument.control(
+        "DSNS?", "DSNS %g",
+        """Control DUT sensor type.
+
+        :type: string
+
+        Possible values are:
+
+        ======  ======
+        String  Meaning
+        ======  ======
+        ''      no DUT
+        'T'     T-DUT
+        'K'     K-DUT
+        ======  ======
+
+        Warning: If in DUT mode without DUT being connected, TS flags DUT error
+
+        """,
+        validator=strict_discrete_set,
+        values={None: 0, 'T': 1, 'K': 2},
+        map_values=True
+    )
+
+    dut_constant = Instrument.control(
+        "DUTC?", "DUTC %g",
+        """Control thermal constant (default 100) of DUT.
+
+        :type: float
+
+        Lower values indicate lower thermal mass, higher values indicate higher
+        thermal mass respectively.
+        """,
+        validator=truncated_range,
+        values=[20, 500]
+    )
+
+    head = Instrument.control(
+        "HEAD?", "HEAD %s",
+        """Control TS head position.
+
+        :type: string
+
+        ``down``: transfer head to lower position
+        ``up``:   transfer head to elevated position
+        """,
+        validator=strict_discrete_set,
+        values={'up': 0, 'down': 1},
+        map_values=True
+    )
+
+    enable_air_flow = Instrument.setting(
+        "FLOW %g",
+        """Set TS air flow.
+
+        ``True`` enables air flow, ``False`` disables it
+
+        :type: bool
+
+        """,
+        validator=strict_discrete_set,
+        values={0, 1, True, False}
```

Comment:
In pymeasure we generally prefer "boolean" properties to expose Bools only (to have a cleaner API), in this pattern: 
```
        validator=strict_discrete_set,
        map_values=True,
        values={True: 1, False: 0},
```
(I put this here, instead of in the other previously resolved conversation because that one seems not nicely visible because it's attached to an outdated source line)